src/plots/cartesian/axes.js

```diff
@@ -2559,10 +2608,29 @@ axes.drawOne = function(gd, ax, opts) {
             }
         }
 
+        var axDepth = 0;
+        var titleDepth = 0;
+        var multAxisPad = 10; // TODO: Expose as a param to allow user to specify padding between axes
+        if(ax._shiftPusher) {
+            if(s === 'l') {
+                axDepth = Math.max(llbbox.height > 0 ? pos - llbbox.left : 0, outsideTickLen);
+            } else {
+                axDepth = Math.max(llbbox.height > 0 ? llbbox.right - pos : 0, outsideTickLen);
+            }
+            if(ax.title.text !== fullLayout._dfltTitle[axLetter]) {
+                titleDepth = (ax._titleStandoff || 0) + (ax._titleScoot || 0);
+                if(s === 'l') {
+                    titleDepth += approxTitleDepth(ax);
+                }
```

Comment:
Wondering why we only need to add this in case of `s === 'l'` ?

Reply:
It took me a while to figure out, but it's because the axis title always gets drawn to the left of a baseline: 

```
t |
i |
t |
l |
e | 
```
which means that its size needs to be accounted for differently between axes on the left of the plot vs those on the right